luda/editor/client/src/timeline/clip/SubtitleClipComponent.ts

```diff
@@ -0,0 +1,261 @@
+  strokeBorderPaint.setStrokeCap(CanvasKit.StrokeCap.Round);
+  strokeBorderPaint.setColor(brighterColor);
+
+  return Translate(
+    {
+      x,
+      y: 0,
+    },
+    [
+      {
+        drawCalls: [
+          {
+            commands: [
+              {
+                type: "path",
+                path: strokePath,
+                paint: strokeBorderPaint,
+              },
+            ],
+          },
+        ],
+      },
+
+      Translate(
+        headPosition,
+        Clip(
+          {
+            path: headPath,
+            clipOp: CanvasKit.ClipOp.Intersect,
+          },
+          {
+            drawCalls: [
+              {
+                commands: [
+                  {
+                    type: "path",
+                    path: headPath,
+                    paint: fillPaint,
+                  },
+                  {
+                    type: "path",
+                    path: headPath,
+                    paint: borderPaint,
+                  },
+                ],
+              },
+            ],
+          },
+        ),
+      ),
+
+      Translate(
+        tailPosition,
+        Clip(
+          {
+            path: tailPath,
+            clipOp: CanvasKit.ClipOp.Intersect,
+          },
+          {
+            drawCalls: [
+              {
+                commands: [
+                  {
+                    type: "path",
+                    path: tailPath,
+                    paint: fillPaint,
+                  },
+                  {
+                    type: "path",
+                    path: tailPath,
+                    paint: borderPaint,
+                  },
+                ],
+              },
+            ],
+          },
+        ),
+      ),
+
+      {
+        drawCalls: [
+          {
+            commands: [
+              {
+                type: "path",
+                path: strokePath,
+                paint: strokeFillPaint,
+              },
+            ],
+          },
+        ],
+      },
+
+      AfterDraw(({ translated }) => {
+        const mouse = engine.mousePosition.mousePosition;
+
+        const mouseInHead = headPath.contains(
+          mouse.x - translated.x - headPosition.x,
+          mouse.y - translated.y - headPosition.y,
+        );
+
+        const mouseInTail = tailPath.contains(
+          mouse.x - translated.x - tailPosition.x,
+          mouse.y - translated.y - tailPosition.y,
+        );
+
+        if (mouseInHead) {
+          engine.mousePointer.setCursor(Cursor.grab);
+        }
+
+        if (mouseInTail) {
+          engine.mousePointer.setCursor(Cursor.leftRightResize);
+        }
+
+        const mouseIn = mouseInHead || mouseInTail;
+
+        if (mouseIn) {
+          timelineState.clipIdMouseIn = clip.id;
+        }
+
+        if (timelineState.clipIdMouseIn === clip.id && !mouseIn) {
+          timelineState.clipIdMouseIn = undefined;
+        }
+
+        engine.mouseEvent.onMouseDown((mouseEvent) => {
+          if (mouseIn) {
```

Comment:
I use `fakeDraggingClip` concept
with https://github.com/NamseEnt/namseent/pull/18/commits/996340e0def18dd72136209d35e57ab311b61ab9

And fixed about mouse event
with https://github.com/NamseEnt/namseent/pull/18/commits/01fdc51d52f0c8fea8d1d879c926e812a1352c9c

Reply:
I super recommend, removing `engine.mouseEvent.onMouseDown((mouseEvent) => {` from AfterDraw and just put this function into renderingData above.  There is head renderingData and tail renderingData.
Before, we share `mouseIn` with outside of event handler, it looks better, but now, we re-calculate `mouseIn`, then I believe there are no reason to put this function in here..
One of big benefit if you move this function in renderingData's OnClick is, you can make it sure that mouse in tail or head because that rendering data are for tail or head.